server/lib/chat/chat-models.ts

```diff
@@ -721,12 +732,15 @@ export async function searchChannelsAsAdmin(
     query.append(sql`
       ORDER BY user_count DESC, name
       LIMIT ${limit}
-      OFFSET ${pageNumber * limit}
+      OFFSET ${offset}
     `)
 
-    const result = await client.query<DbChannel>(query)
+    const result = await client.query<DbChannel & { joined: boolean }>(query)
 
-    return result.rows.map(row => convertChannelFromDb(row))
+    return {
+      channels: result.rows.map(row => convertChannelFromDb(row)),
+      joinedChannels: result.rows.filter(row => row.joined).map(row => row.id),
```

Comment:
We go to a bunch of extra effort to basically join this up, such that the list of joined channels is only what is in the returned list, but this doesn't actually feel useful in any way? I think it'd be better to remove this functionality from this method entirely and just send 2 queries to the DB: one to do the search and one to get the list of joined channels. If you want a function that returns just the joined channel IDs without any info, that's fine to create (and very simple)